### PR TITLE
Update wrong catch block

### PR DESCRIPTION
**Pull request by liviarett**
##### Checklist

- [ ] `npm install && npm test` passes
- [ ] tests are included 
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/googleapis/release-please#how-should-i-write-my-commits)

### CODE REVIEW

**Comment by cclauss** [comment]
https://stackoverflow.com/questions/3335376/why-do-good-programmers-sometimes-silently-swallow-exceptions

**Comment by rvagg** [comment]
Yeah, sorry, this is intentional (IIRC), there's some OS' where this fails, I can't remember the details, it was some trimmed-down Linux, I'm sure `git blame` would point the way to the PR that introduced the quiet fail.

Is this causing problems for you?

**Comment by liviarett** [comment]
hmm I had a few other issues that were causing me to get to this point anyway, but this is disguising the error, as it gives me 

```
gyp ERR! UNCAUGHT EXCEPTION
gyp ERR! stack /Users/liviarett/.nvm/versions/node/v8.17.0/lib/node_modules/node-gyp/lib/find-python.js:44
gyp ERR! stack   } catch {}
gyp ERR! stack           ^
gyp ERR! stack
gyp ERR! stack SyntaxError: Unexpected token {
gyp ERR! stack     at createScript (vm.js:80:10)
gyp ERR! stack     at Object.runInThisContext (vm.js:139:10)
gyp ERR! stack     at Module._compile (module.js:617:28)
gyp ERR! stack     at Object.Module._extensions..js (module.js:664:10)
gyp ERR! stack     at Module.load (module.js:566:32)
gyp ERR! stack     at tryModuleLoad (module.js:506:12)
gyp ERR! stack     at Function.Module._load (module.js:498:3)
gyp ERR! stack     at Module.require (module.js:597:17)
gyp ERR! stack     at require (internal/module.js:11:18)
gyp ERR! stack     at Object.<anonymous> (/Users/liviarett/.nvm/versions/node/v8.17.0/lib/node_modules/node-gyp/lib/configure.js:11:18)
```

but not a big problem :)

File: lib/find-python.js [hunk 1]
```diff
@@ -36,7 +36,9 @@ for (const majorMinor of ['39', '38', '37', '36']) {
 function getOsUserInfo () {
   try {
     return require('os').userInfo().username
-  } catch {}
+  } catch (e) {
+    throw e
```
Comment (rvagg) [inline lib/find-python.js:40]:
```suggestion
```

**Review by rvagg — CHANGES_REQUESTED** [review]
ookkaayy ... my memory is failing me on this one, I see this particular change was introduced @ fca4795512c67dc8420aaa0d913b5b89a4b147f3 and that I must be thinking of another silent catch that we introduced eons ago.. So this is pretty new, and a legit parse error.

:thumbsup: so let's fix the parse error you're getting, but not force the throw.

**Comment by rvagg** [comment]
ahha! I knew I wasn't going _that_ senile: https://github.com/nodejs/node-gyp/pull/1835 but it's in install.js, while this same code is now in find-python.js which was just recently refactored.

**Comment by liviarett** [comment]
ah alright! nice, so we can at least fix the unexpected token thing

**Comment by DeeDeeG** [comment]
@cclauss

The `catch {}` syntax is valid on our supported versions of Node (10.12 and newer). I honestly hadn't anticipated this would every run on Node older than that, but in hindsight that's a little optimistic.

The `catch {}` I added in https://github.com/nodejs/node-gyp/pull/2375 is admittedly a syntax error on older Node. Older node wants the error to be passed into the catch block: `catch (e) {}`.

[MDN says](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#browser_compatibility) this is  a change as of Node 10/Chrome 66, but I'm finding it hard to confirm that anywhere in NodeJS or Chrome's documentation/blogs. Maybe this NodeJS PR? https://github.com/nodejs/node/pull/20708 (landed for Node 10.2 and newer).

I tried logging the error (https://github.com/nodejs/node-gyp/pull/2375#issuecomment-818217760 --> [implementation](https://github.com/DeeDeeG/node-gyp/compare/master...DeeDeeG:fail-gracefully-winDefaultLocations-bigger-refactor-with-logging-and-comments#diff-389dd7e63af69fe94fc71e79916ed76b0114dfdf49efe26112a13efa9e6531f6R349-R359)), but that added complexity and size of code seemed a bit overkill, given we intend to fail gracefully and just move on. It is hard to properly add logging at file-level scope (outside of `PythonFinder`), and I found refactoring this into `PythonFinder` tricky or impossible, since this data needs to be computed for `winDefaultLocations` to exist when `PythonFinder` is made... That's as well as I know how to explain it, given my limited JS proficiency. (I am self-taught with JS, and `PythonFinder` honestly confuses me a bit.)

**Review by DeeDeeG — APPROVED** [review]
:+1: Avoids a syntax error on older Node (Node 9.x or older).

_(Note: Those versions of Node are technically not supported by the latest `node-gyp`, because of https://github.com/nodejs/node-gyp/pull/2123 --> https://github.com/nodejs/node-gyp/issues/2152. But I think not causing a syntax error in the program for older Node is still an improvement.)_

Due to https://github.com/nodejs/node-gyp/issues/2152, users should really be on Node 10.12 or newer when using the latest `node-gyp`, and ideally they would be on the latest Node 12.x or 14.x or 16.x. But I don't think fixing this syntax error for older Node (at the cost of three characters of JS) hurts anything.

**Comment by rvagg** [comment]
:thumbsup: thanks @DeeDeeG, it's good to lean on the conservative side for node-gyp since it's in such wide usage. When people show up using older versions of Node.js (for whatever reason - sometimes they can't change) and well tell them that their old version of node-gyp doesn't work for their use-case, it's not fun to tell them they can't get the newer one cause their Node.js is old. So where the change is painless, like this one, let's opt for the conservative choice.

**Comment by DeeDeeG** [comment]
Update: I found out that trying to use `node-gyp` 8 on older Node (Node < 10) is still rather busted, due to https://github.com/nodejs/node-gyp/pull/2220.

Here's the details and a workaround...

https://github.com/nodejs/node-gyp/pull/2220/ broke `node-gyp install` (and, by extension, broke `node-gyp rebuild`) on Node older than version 10, by relying on the [`stream.pipeline` Node API function](https://nodejs.org/api/stream.html#stream_stream_pipeline_streams_callback) that was only added in Node v10.0.0. And by switching a bunch of `fs.x` to `fsPromises.x`. (The `fsPromises` API was also added in Node 10.0.0).

<details><summary>Workaround if you really need to run the latest <code>node-gyp</code> on Node 8.x would be:</summary>

Caveat/warning: All this only applies to `node-gyp`@8.1.0. Only tested somewhat working by running `node-gyp rebuild` via `npm rebuild`... Using `npm` 6.13.4 and Node 8.17.0. Only tested on Linux. Any other combination may not work, and none of this is actually officially supported by `node-gyp` or `npm` or Node, etc. Try at your own risk.

- Install older Node's development headers with an older `node-gyp` that works on older Node, such as Node 8.x.
  - (With the older Node version, run `npx node-gyp@6 install`)
- Then, go in with a text editor and manually edit the copy of `node-gyp` you want to run; Edit `node-gyp`'s `lib/find-python.js` file like this PR does; update the `catch {}` block in `lib/find-python.js` to pass in the error: `catch (e) {}`.
  - (Patch the file [here](https://github.com/nodejs/node-gyp/blob/v8.1.0/lib/find-python.js#L39).)
- Then in `lib/install.js`...
  -  Go in with a text editor and manually replace `const streamPipeline = util.promisify(stream.pipeline)` near the top of the file with an empty function: `const streamPipeline = function () {}`.
      - ([here](https://github.com/nodejs/node-gyp/blob/v8.1.0/lib/install.js#L15).)
  - And replace `fs.promises.stat()` with `fs.statSync()`, and replace `fs.promises.readFile()` with `fs.readFileSync()`, and delete the `await` immediately preceding these two `fs` API calls, since we're using basic synchronous code now, not Promises.
    - (Only needed in two places. [here](https://github.com/nodejs/node-gyp/blob/v8.1.0/lib/install.js#L57) and [here](https://github.com/nodejs/node-gyp/blob/v8.1.0/lib/install.js#L75).)

</details>

**Pull request merged**